internal/server/controller.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package server
+
+import "github.com/hashicorp/boundary/internal/server/store"
+
+// Controllers are a server that provider user auth to a client and contain the permissions and resources.
+// Additionally controllers communicate with external services such as databases, KMS, idp, plugins, etc.
```

Comment:
I think some of these details in this doc are quite correct. For example a lot fo that communication to external services can actually happen via workers,
but the controller tells the worker to make the request. Perhaps instead something like:

```suggestion
// Controller is a server that is responsible for understanding configuration,
// authenticating and authorizing users, and serving user API requests (e.g. to
// initiate a session). They also assign tasks to workers (session handling,
// session recording parsing, etc.).
```

Reply:
Good callout, went ahead and updated the comment.